File: gestion-encuestas-backend/src/main/java/com/fullstack/encuestas/services/EncuestaService.java
package com.fullstack.encuestas.services;

import com.fullstack.encuestas.entities.Encuesta;
import com.fullstack.encuestas.entities.Pregunta;

import java.util.List;
import java.util.Optional;

public interface EncuestaService {

    List<Encuesta> getAllEncuestas();
    Optional<Encuesta> getEncuestaById(Long id);

    Encuesta createEncuesta(Encuesta encuesta);

    Encuesta updateEncuesta(Long id, Encuesta encuesta);

    void deleteEncuesta(Long id);

}

File: gestion-encuestas-frontend/src/app/services/encuesta.service.ts
import { Injectable } from '@angular/core';
import { Encuesta } from '../models/encuesta';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class EncuestaService {

  private baseUrl: string = 'http://localhost:8080/api/encuestas';

  constructor(private httpClient: HttpClient) {}

  getAllEncuestas(): Observable<Encuesta[]> {
    return this.httpClient.get<Encuesta[]>(this.baseUrl);
  }

  getEncuestaById(id: number): Observable<Encuesta> {
    return this.httpClient.get<Encuesta>(`${this.baseUrl}/${id}`);
  }

  saveEncuesta(encuesta: Encuesta): Observable<Encuesta> {
    return this.httpClient.post<Encuesta>(this.baseUrl, encuesta);
  }

  updateEncuesta(id: number, encuesta: Encuesta): Observable<Encuesta> {
    return this.httpClient.put<Encuesta>(`${this.baseUrl}/${id}`, encuesta);
  }

  deleteEncuesta(id: number): Observable<void> {
    return this.httpClient.delete<void>(`${this.baseUrl}/${id}`);
  }
}
